src/axmysql/sagarepository.go

```diff
@@ -0,0 +1,158 @@
+package axmysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/jmalloc/ax/src/ax/persistence"
+	"github.com/jmalloc/ax/src/ax/saga"
+)
+
+// SagaRepository is an implementation of crud.Repository that stores saga
+// instances using SQL persistence.
+type SagaRepository struct{}
+
+// LoadSagaInstance fetches a saga instance that has a specific mapping key
+// in its key set.
+//
+// sn is the saga name. k is the message mapping key.
+//
+// If a saga instance is found; ok is true, otherwise it is false. A
+// non-nil error indicates a problem with the store itself.
+//
+// It panics if the repository is not able to enlist in tx because it uses a
+// different underlying storage system.
+func (r SagaRepository) LoadSagaInstance(
+	ctx context.Context,
+	ptx persistence.Tx,
+	id saga.InstanceID,
+) (saga.Instance, error) {
+	var (
+		i           saga.Instance
+		contentType string
+		data        []byte
+	)
+
+	if err := sqlTx(ptx).QueryRowContext(
+		ctx,
+		`SELECT
+			instance_id,
+			revision,
+			content_type,
+			data
+		FROM saga_instance
+		WHERE instance_id = ?`,
+		id,
+	).Scan(
+		&i.InstanceID,
+		&i.Revision,
+		&contentType,
+		&data,
+	); err != nil {
+		return saga.Instance{}, err
+	}
+
+	var err error
+	i.Data, err = saga.UnmarshalData(contentType, data)
+
+	return i, err
+}
+
+// SaveSagaInstance persists a saga instance and its associated mapping
+// table to the store as part of tx.
+//
+// It returns an error if the saga instance has been modified since it was
```

Comment:
Isn't the whole idea to save modifications?

Reply:
Yeah that's poorly worded, it should say "modified by another process" or something like that.